plans/super/139-snowflake-sample-shape.md

```
@@ -0,0 +1,255 @@
+    dialect: Dialect,
+    sample_bucket: int,
+    sample_size: int,
+    extra_where: str | None = None,   # pre-rendered partition predicate (caller owns it)
+    order_by_hash: bool,              # compiler CTE: False; adapters: True
+) -> str: ...
+```
+
+Reads `dialect.sample_row_hash_expr`, `dialect.sample_hash_in_projection`, `dialect.sample_hash_alias`. Two branches, switched on the **boolean** (never `dialect.name`):
+
+- **Inline (BigQuery, `sample_hash_in_projection=False`)** — byte-identical to today's compiler CTE body / would-be adapter form:
+  ```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)